src/components/DataTable/TableExpandedRow.js

```diff
@@ -7,20 +7,48 @@
 
 import cx from 'classnames';
 import PropTypes from 'prop-types';
-import React from 'react';
+import React, { useRef } from 'react';
 import { settings } from 'carbon-components';
+import TableCell from './TableCell';
 
 const { prefix } = settings;
 
 const TableExpandedRow = ({
   className: customClassName,
   children,
+  colSpan,
   ...rest
 }) => {
-  const className = cx(`${prefix}--expandable-row-v2`, customClassName);
+  const rowRef = useRef(null);
+  const className = cx(`${prefix}--expandable-row`, customClassName);
+
+  const onMouseEnter = () => {
+    if (rowRef && rowRef.current) {
```

Comment:
Any reason why you did direct DOM access instead of state update then render?

Reply:
I tried a couple options, I found this one the easiest to reason about and simplest. 

Since the class is toggled on a sibling dom node, state would have to be pulled up to the common parent component. 

You’d then have to do introspection on the children to find the hovered element and it’s sibling. This would also  force a rerender of the whole table every time a new row is hovered. 

My thought was that until we need to perform other actions based on the hover state, introducing new state just to track the hover event unnecessarily complicated the component. 

Another alternative I considered was introducing a wrapper for the Row and Expanded Row pairs but the nesting structure of our table styles and the current amount of table components made this seem unwise for just this one usecase.

Reply:
Thanks @vpicone for elaborating what's behind this approach @vpicone! - Which made me realize that the code here changes DOM of other component, assuming such other components exists at a certain location in DOM relative to this component. We need to make our codebase robust considering future change in such DOM location - AFAIK this is one of the biggest reasons React preaches for avoiding direct DOM access wherever possible.

Making the fragment containing `<TableExpandRow>` and `<TableExpandedRow>` a component looked a most straightforward option, though we need support for event handlers on fragments (can be seen at issue# 12051 in React repo).

If we still want to go with direct DOM access approach, I'd recommend at walking up through the siblings to find one matching `.${prefix}--expandable-row` selector.

Reply:
the `TableExpandedRow` has many siblings with the `--parent-row` selector. Many of those parent rows might also have `--expanded-row` (meaning they are currently expanded). There could also be hundreds of rows each with an adjacent expanded rows and this gets fired onHover.

Querying for a specific css selector seems brittle in a different way. I could be wrong here, but I believe refactoring a className shouldn't break the behavior of the component. I think my assumption that the TableExpandedRow has a sibling parent row may be more robust than it having a that sibling and that sibling also containing a specific className.

I mentioned a wrapper component, maybe `ExpandableRow`, but I agree that it would make an already difficult API even more so.

I could add a check to ensure that the previousSibling is a `--parent-row` but that gets back to relying on classNames for behavior.

Reply:
@vpicone Change in class name is a great point. Another possible approach is marking parent row `<tr>` with an (`data-`) attribute and searching for that with an attribute selector. BTW when I talk about walking up through the siblings, I mean using the first one found, not about finding all matching. DOM element has `.matches(selector)` API where you can see an element matches the given selector or not.

Reply:
How would we mark the parent row without already having a mark on the parent row? Finding it to mark it has the same issues as finding it to style it.

We could have a required key prop but at that point we minus well just have a wrapper component.

Reply:
I was going to talk about just adding `data-` attribute on the parent row `<tr>`, but seems that we have one already: https://github.com/IBM/carbon-components-react/blob/3d6ec01297638d9a1d3b2ae6e537074b779defa8/src/components/DataTable/TableExpandRow.js#L39

Reply:
So do you think adding a check for that custom data attribute on the previousSibling is necessary/less brittle?

Reply:
Yeah - And again we should walk through the siblings until we find one.